buildSrc/src/main/java/io/micronaut/guides/core/MacroUtils.java

```diff
@@ -0,0 +1,80 @@
+package io.micronaut.guides.core;
+
+import groovy.transform.Memoized;
+import io.micronaut.core.annotation.NonNull;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static io.micronaut.guides.core.GuidesConfigurationProperties.LICENSEHEADER;
+
+public class MacroUtils {
+    public static String extractName(String line, String macro) {
+        return line.substring(macro.length()+1, line.indexOf('['));
+    }
+
+    public static String extractAppName(String line) {
+        return extractFromParametersLine(line, "app");
+    }
+
+    public static List<String> extractTags(String line) {
+        String attributeValue = extractFromParametersLine(line, "tags");
+        if (attributeValue != null && !attributeValue.isEmpty()) {
+            return Arrays.asList(attributeValue.split("\\|"));
+        }
+
+        return extractTagName(line).isEmpty() ? Collections.emptyList() : Collections.singletonList(extractTagName(line));
+    }
+
+    @NonNull
+    public static String mainPath(@NonNull String appName,
+                           @NonNull String fileName, GuidesOption option) {
+        return pathByFolder(appName, fileName, "main", option);
+    }
+
+    @NonNull
+    private static String pathByFolder(@NonNull String appName,
+                                       @NonNull String fileName,
+                                       String folder, GuidesOption option) {
+        String module = appName != "" ? appName + "/" : "";
+        return module+"src/"+folder+"/"+option.getLanguage().toString()+"/example/micronaut/"+fileName+"."+option.getLanguage().getExtension();
+    }
+
+    public static String extractIndent(String line) {
+        String indentValue = extractFromParametersLine(line, "indent");
+        return indentValue != "" ? "indent="+indentValue : "";
+    }
+
+    @Memoized
```

Comment:
[@Memoized](https://docs.groovy-lang.org/2.4.9/html/gapi/groovy/transform/Memoized.html) is an Apache Groovy AST transformation it will not work in Groovy. 
```suggestion

```

Reply:
I have removed this